### ISSUE TITLE
Config opencv support cuda environment

### ISSUE THREAD

**Issue by RichardTamarisk** [opencv/opencv#26392]
### System Information

OpenCV version: 4.8.1
Operating System / Platform: ubuntu 22.04
compiler: gcc 11.4.0
Cuda compilation tools, release: 12.6
FFmpeg version: n5.1.4
opencv: 4.8.1
opencv_contrib: 4.8.1

### Detailed description

<img width="452" alt="image" src="https://github.com/user-attachments/assets/8f14571b-267a-4c7d-bc59-c44fe6aaf52d">
<img width="462" alt="image" src="https://github.com/user-attachments/assets/a345cbd6-02fb-4e41-8280-9d1c4be07e48">
<img width="567" alt="image" src="https://github.com/user-attachments/assets/600df4d0-bd5c-4a0b-bfeb-46c847363f4e">

Above was I used the command:
cd opencv
mkdir build 
cmake -D WITH_CUDA=ON -D CUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda -D OPENCV_EXTRA_MODULES_PATH=../../opencv_contrib/modules ..
make

### Steps to reproduce

The full process:
git clone https://github.com/opencv/opencv_contrib.git
git clone https://github.com/opencv/opencv.git
cd opencv
mkdir build
cd build
cmake -D WITH_CUDA=ON -D CUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda -D OPENCV_EXTRA_MODULES_PATH=../../opencv_contrib/modules ..
make

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [ ] I updated to the latest OpenCV version and the issue is still there
- [ ] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by cudawarped** [comment]
We need to see the full error's not screen shots, but from your build information my guess is the error is related to you using OpenCV 4.8.1 with CUDA 12.6.

To use CUDA 12.6 you need to use the latest commits from the 4.x branch of both the main and contrib repository.

**Comment by RichardTamarisk** [comment]
> We need to see the full error's not screen shots, but from your build information my guess is the error is related to you using OpenCV 4.8.1 with CUDA 12.6.
> 
> To use CUDA 12.6 you need to use the latest commits from the 4.x branch of both the main and contrib repository.

yes, I use latest commits from the 4.8.1 branch of both the main and contrib repository.

**Comment by cudawarped** [comment]
> > We need to see the full error's not screen shots, but from your build information my guess is the error is related to you using OpenCV 4.8.1 with CUDA 12.6.
> > To use CUDA 12.6 you need to use the latest commits from the 4.x branch of both the main and contrib repository.
> 
> yes, I use latest commits from the 4.8.1 branch of both the main and contrib repository.

4.x branch https://github.com/opencv/opencv/branches not 4.8.1 Tag https://github.com/opencv/opencv/tags.  OpenCV 4.8.1 was [released September 2023](https://github.com/opencv/opencv/wiki/ChangeLog#version481) and CUDA 12.6 was [released in August 2024](https://developer.nvidia.com/cuda-toolkit-archive)

**Comment by RichardTamarisk** [comment]
> > > We need to see the full error's not screen shots, but from your build information my guess is the error is related to you using OpenCV 4.8.1 with CUDA 12.6.
> > > To use CUDA 12.6 you need to use the latest commits from the 4.x branch of both the main and contrib repository.
> > 
> > 
> > yes, I use latest commits from the 4.8.1 branch of both the main and contrib repository.
> 
> 4.x branch https://github.com/opencv/opencv/branches not 4.8.1 Tag https://github.com/opencv/opencv/tags. OpenCV 4.8.1 was [released September 2023](https://github.com/opencv/opencv/wiki/ChangeLog#version481) and CUDA 12.6 was [released in August 2024](https://developer.nvidia.com/cuda-toolkit-archive)

[ 14%] Building CXX object 3rdparty/openexr/CMakeFiles/IlmImf.dir/Imath/ImathVec.cpp.o
[ 14%] Linking C static library ../lib/liblibwebp.a
[ 14%] Built target libwebp
[ 14%] Linking CXX static library ../lib/libIlmImf.a
[ 14%] Built target IlmImf
[ 14%] Linking CXX static library ../lib/liblibprotobuf.a
[ 14%] Built target libprotobuf
make[2]: *** [modules/core/CMakeFiles/opencv_core.dir/build.make:77: modules/core/CMakeFiles/cuda_compile_1.dir/src/cuda/cuda_compile_1_generated_gpu_mat.cu.o] Interrupt
make[1]: *** [CMakeFiles/Makefile2:4071: modules/core/CMakeFiles/opencv_core.dir/all] Interrupt
make: *** [Makefile:166: all] Interrupt

I used command:``` make  >  build_log.txt 2>&1 ```got above information that I hope it can provide you with more useful information.

**Comment by cudawarped** [comment]
> Can you offer opencv versions that compatible with both FFmpeg n5.1.4 and CUDA 12.6?

Yes the latest commits from the 4.x branches.  The last version 4.10 had fixes applied after it was released to accomodate CUDA 12.6, e.g. https://github.com/opencv/opencv_contrib/pull/3785

I would suggest using CUDA 12.4 although I can't see why you need to use OpenCV 4.8.1 with FFmpeg 5.1.4, I use the latest commits on the 4.x branches with FFmpeg 5.0 without any issues.

**Comment by RichardTamarisk** [comment]
> > Can you offer opencv versions that compatible with both FFmpeg n5.1.4 and CUDA 12.6?
> 
> Yes the latest commits from the 4.x branches. The last version 4.10 had fixes applied after it was released to accomodate CUDA 12.6, e.g. [opencv/opencv_contrib#3785](https://github.com/opencv/opencv_contrib/pull/3785)
> 
> I would suggest using CUDA 12.4 although I can't see why you need to use OpenCV 4.8.1 with FFmpeg 5.1.4, I use the latest commits on the 4.x branches with FFmpeg 5.0 without any issues.

Thank you very much for your reply!!! I will try the version you proposed.
